comp/publishermetadatacache/impl/publishermetadatacache_test.go

```diff
@@ -0,0 +1,148 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+//go:build windows
+
+package publishermetadatacacheimpl
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	publishermetadatacache "github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/publishermetadatacache"
+
+	evtapi "github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/api"
+	fakeevtapi "github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/api/fake"
+)
+
+func TestPublisherMetadataCache_Get(t *testing.T) {
+	cache := publishermetadatacache.New(fakeevtapi.New())
+
+	publisherName1 := "Publisher1"
+	publisherName2 := "Publisher2"
+
+	handle1, err := cache.Get(publisherName1)
+	assert.NoError(t, err)
+	handle2, err := cache.Get(publisherName2)
+	assert.NoError(t, err)
+
+	assert.NotEqual(t, publishermetadatacache.InvalidHandle, handle1)
+	assert.NotEqual(t, publishermetadatacache.InvalidHandle, handle2)
+
+	// Verify item is in cache
+	cachedValue, found := cache.GetCache().Load(publisherName1)
+	assert.True(t, found)
+	assert.Equal(t, handle1, cachedValue.(publishermetadatacache.CacheEntry).Handle)
+
+	// Verify item is in cache
+	cachedValue, found = cache.GetCache().Load(publisherName2)
+	assert.True(t, found)
+	assert.Equal(t, handle2, cachedValue.(publishermetadatacache.CacheEntry).Handle)
+}
+
+func TestPublisherMetadataCache_FormatMessage_InvalidHandle_RecreatesCache(t *testing.T) {
+	fakeAPI := fakeevtapi.New()
+	cache := publishermetadatacache.New(fakeAPI)
+
+	publisherName := "TestPublisher"
+	eventHandle := evtapi.EventRecordHandle(100)
+
+	originalHandle, err := cache.Get(publisherName)
+	assert.NoError(t, err)
+	assert.NotEqual(t, publishermetadatacache.InvalidHandle, originalHandle)
+
+	// Invalidate the handle to simulate provider being uninstalled
+	err = fakeAPI.InvalidatePublisherHandle(originalHandle)
+	assert.NoError(t, err)
+
+	// FormatMessage should detect invalid handle and remove from cache
+	message, err := cache.FormatMessage(publisherName, eventHandle, 0)
+	assert.Empty(t, message)
+	assert.Contains(t, err.Error(), "not implemented") // Fake API returns error
```

Comment:
what does invalidating the handle do here if FormatMessage always returns "not implemented error" ?